internal/konnect/node_agent.go

```diff
@@ -0,0 +1,118 @@
+package konnect
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/util"
+)
+
+var defaultRefreshNodeInterval = 15 * time.Second
+
+type NodeAgent struct {
+	NodeID   string
+	Hostname string
+	Version  string
+
+	Logger logr.Logger
+
+	adminClient     *AdminClient
+	refreshInterval time.Duration
+}
+
+func NewNodeAgent(hostname string, version string, logger logr.Logger, adminClient *AdminClient) *NodeAgent {
+	return &NodeAgent{
+		Hostname: hostname,
+		Version:  version,
+		Logger: logger.
+			WithName("konnect-node").WithValues("runtime_group_id", adminClient.RuntimeGroupID),
+		adminClient: adminClient,
+		// TODO: set refresh interval by flags/envvar
+		refreshInterval: defaultRefreshNodeInterval,
+	}
+}
+
+func (a *NodeAgent) createNode() error {
+	createNodeReq := &CreateNodeRequest{
+		ID:       a.NodeID,
+		Hostname: a.Hostname,
+		Version:  a.Version,
+		Type:     NodeTypeIngressController,
+		LastPing: time.Now().Unix(),
+	}
+	resp, err := a.adminClient.CreateNode(createNodeReq)
+	if err != nil {
+		a.Logger.Error(err, "failed to create node")
+		return err
```

Comment:
Perhaps create a wrapper error instead?

```suggestion
		return fmt.Errorf("failed to create node: %w", err)
```

Reply:
done